common/buildcraft/factory/tile/TileMiner.java

```diff
@@ -0,0 +1,193 @@
+package buildcraft.factory.tile;
+
+import buildcraft.api.mj.MjBattery;
+import buildcraft.api.tiles.IControllable;
+import buildcraft.api.tiles.IDebuggable;
+import buildcraft.api.tiles.IHasWork;
+import buildcraft.lib.migrate.BCVersion;
+import buildcraft.lib.misc.MessageUtil;
+import buildcraft.lib.tile.TileBC_Neptune;
+import net.minecraft.block.Block;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.ITickable;
+import net.minecraft.util.math.BlockPos;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+
+import java.io.IOException;
+import java.util.List;
+
+public abstract class TileMiner  extends TileBC_Neptune implements ITickable, IHasWork, IControllable, IDebuggable {
+    public static final int NET_LED_STATUS = 10;
+
+    protected int progress = 0;
+    protected BlockPos currentPos = null;
+    // Used to check if this has completed all work
+    protected boolean isComplete = false;
+    protected Mode mode = Mode.On;
+    protected final MjBattery battery = new MjBattery(1000_000);
+
+    protected void initCurrentPos() {
+        if (currentPos == null) {
+            currentPos = getPos();
+            currentPos = currentPos.down();
+        }
+    }
+
+    protected abstract void mine();
+
+    protected abstract Block getBlockForDown();
+
+    @Override
+    public void update() {
+        if (worldObj.isRemote || mode != Mode.On) {
+            return;
+        }
+        battery.tick(getWorld(), getPos());
+
+        // test with the output of a stone engine
+        battery.addPower(1000);// remove this
+
+        if (worldObj.rand.nextDouble() > 0.9) { // is this correct?
```

Comment:
No, that was temporary for testing. It needs to be replaced with a check to see if it should actually send an update message yet (the status of any of the LED's has changed)